src/tests/api/test_subevents.py

```diff
@@ -50,6 +109,242 @@ def test_subevent_list(token_client, organizer, event, subevent):
+    assert resp.content.decode() == '{"meta_data":["Meta data property \'foo\' does not exist."]}'
+
+    resp = token_client.post(
+        '/api/v1/organizers/{}/events/{}/subevents/'.format(organizer.slug, event.slug),
+        {
+            "name": {
+                "de": "Demo Subevent 2020 Test",
+                "en": "Demo Subevent 2020 Test"
+            },
+            "active": False,
+            "date_from": "2017-12-27T10:00:00Z",
+            "date_to": "2017-12-28T10:00:00Z",
+            "date_admission": None,
+            "presale_start": None,
+            "presale_end": None,
+            "location": None,
+            "item_price_overrides": [
+                {
+                    "item": item.pk,
+                    "price": "23.42"
+                }
+            ],
+            "variation_price_overrides": [],
+            "meta_data": {
+                "type": "Workshop"
+            },
+        },
+        format='json'
+    )
+    assert resp.status_code == 201
+    assert organizer.events.get(slug="dummy").subevents.get(id=resp.data['id']).items.get(id=item.pk).default_price == Decimal('23.00')
+    assert organizer.events.get(slug="dummy").subevents.get(id=resp.data['id']).item_price_overrides[item.pk] == Decimal('23.42')
+
+    resp = token_client.post(
+        '/api/v1/organizers/{}/events/{}/subevents/'.format(organizer.slug, event.slug),
+        {
+            "name": {
+                "de": "Demo Subevent 2020 Test",
+                "en": "Demo Subevent 2020 Test"
+            },
+            "active": False,
+            "date_from": "2017-12-27T10:00:00Z",
+            "date_to": "2017-12-28T10:00:00Z",
+            "date_admission": None,
+            "presale_start": None,
+            "presale_end": None,
+            "location": None,
+            "item_price_overrides": [
+                {
+                    "item": 555,
+                    "price": "23.42"
+                }
+            ],
+            "variation_price_overrides": [],
+            "meta_data": {
+                "type": "Workshop"
+            },
+        },
+        format='json'
+    )
+    assert resp.status_code == 400
+    assert resp.content.decode() == '{"item_price_overrides":[{"item":["Invalid pk \\"555\\" - object does not exist."]}]}'
+
+
+@pytest.mark.django_db
+def test_subevent_update(token_client, organizer, event, subevent, item, meta_prop):
+    resp = token_client.patch(
+        '/api/v1/organizers/{}/events/{}/subevents/{}/'.format(organizer.slug, event.slug, subevent.pk),
+        {
+            "date_from": "2018-12-27T10:00:00Z",
+            "date_to": "2018-12-28T10:00:00Z",
+        },
+        format='json'
+    )
+    assert resp.status_code == 200
+    event = Event.objects.get(organizer=organizer.pk, slug=event.slug)
+    assert organizer.events.get(slug=event.slug).subevents.get(id=resp.data['id']).meta_values.filter(
+        property__name=meta_prop.name, value="Workshop"
+    ).exists()
```

Comment:
What exactly is this first part testing? It's setting dates – and then checking for meta information.